include/zephyr/drivers/uart_emul.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_UART_EMUL_H_
+#define ZEPHYR_INCLUDE_DRIVERS_UART_EMUL_H_
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/sys/slist.h>
+#include <zephyr/types.h>
+
+/**
+ * @file
+ *
+ * @brief Public APIs for the UART device emulation drivers.
+ */
+
+/**
+ * @brief UART Emulation Interface
+ * @defgroup uart_emul_interface UART Emulation Interface
+ * @ingroup io_emulators
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct uart_emul_device_api;
+
+/**
+ * @brief Define the emulation callback function signature for
+ * uart_emul_device_tx_data_ready_set() function.
+ *
+ * @param dev UART device instance
+ * @param size Number of available bytes in TX buffer
+ * @param user_data Arbitrary user data
```

Comment:
I think changing the `@param` argument should fix the doc build error. Just a suggestion for the description. Feel free to use something better.

```suggestion
 * @param emul pointer to emulation context
```